doc/neps/nep-0034.rst

```diff
@@ -0,0 +1,87 @@
+==========================================
+NEP 34 — Disallow default ``dtype=object``
+==========================================
+
+:Author: Matti Picus
+:Status: Draft
+:Type: Standards Track
+:Created: 2019-10-10
+
+
+Abstract
+--------
+
+``np.array([<value])`` with no ``dtype`` keyword argument will sometimes
+default to an ``object``-dtype array. Change the behaviour to raise a
+`ValueError` instead.
```

Comment:
I think this should be clear if this is intended for all cases. For instance, is `np.array([None, None])` still allowed? `np.array([Decimal(1), Decimal(2)])`?

Reply:
changed to "Disallow inferring ``dtype=object`` from sequences" and changed a few other places to make the intention clearer